src/components/ContractPanel.tsx

```diff
@@ -0,0 +1,116 @@
+import React, { useState } from "react";
+import { Mesh } from "../contracts/Mesh";
+import { Button, MintModal, BurnModal } from "../components";
+import { shortenAddress } from "../utils/shortenAddress";
+
+type Props = {
+  contract: Mesh;
+  address: string;
+};
+
+const ContractPanel: React.FC<Props> = (props) => {
+  const [authed, setAuthed] = useState(false);
+  const [error, setError] = useState<string>();
+  const [loading, setLoading] = useState(false);
+
+  const [isMintModalOpen, setIsMintModalOpen] = useState(false);
+  const [isBurnModalOpen, setIsBurnModalOpen] = useState(false);
+
+  const { contract, address } = props;
+
+  const authenticate = () => {
+    try {
+      const balance = contract.balanceOf(address);
+
+      setLoading(true);
+
+      balance.then((result) => {
+        if (+result < 1) return setError("No meshies owned by your address!");
+        setError(undefined);
+        return setAuthed(true);
+      });
+    } catch (err: any) {
+      setError(err as string);
+      return setAuthed(false);
+    } finally {
+      setLoading(false);
+    }
+  };
+
+  const logout = () => {
+    setAuthed(false);
+  };
+
+  const mintModalOpen = () => {
+    setIsMintModalOpen(true);
+  };
+
+  const burnModalOpen = () => {
+    setIsBurnModalOpen(true);
+  };
```

Comment:
typically, since you can't have both a mint and burn modal at the same time, you would have `setModalOpen: (status: "mint" | "burn" | "none") => void`

Reply:
no need to change this, though

Reply:
alternatively, just going to set the buttons to open their respective state, rather than calling a function that does the same thing